pkg/csi/cinder/controllerserver.go

```diff
@@ -274,14 +297,21 @@ func (cs *controllerServer) ControllerPublishVolume(ctx context.Context, req *cs
 func (cs *controllerServer) ControllerUnpublishVolume(ctx context.Context, req *csi.ControllerUnpublishVolumeRequest) (*csi.ControllerUnpublishVolumeResponse, error) {
 	klog.V(4).Infof("ControllerUnpublishVolume: called with args %+v", protosanitizer.StripSecrets(*req))
 
+	// Volume cloud
+	volCloud := req.GetSecrets()["cloud"]
```

Comment:
this change is breaking? Means that existing storageClasses does not work? Everyone should now add new parameters to their classes?

Reply:
There is no breaking change.
You could upgrade from last version to this PR version without configuration changes, it is backward compatible.
I use it to migrate my already created PVC on old SC without secrets reference on my new SC. Old PVC still works and new created PVC on old SC without secrets reference still works too.
GetSecrets() method return type `map[string]string` accessing to an undefined key of this kind of map return an empty string `""`.
Simple example to run locally:
```golang
package main

import "fmt"

func main() {
    secrets := make(map[string]string)
    fmt.Println(secrets["cloud"])
    fmt.Println(secrets["cloud"] == "")
}
```
```bash
$ go run ./main.go

true
```
and empty string in this `volCloud` variable refer to the actual configuration cloud.conf section `[Global]`  (section without subsection or section with subsection `""` is the same behavior)